tests/collections/collection.test.ts

```diff
@@ -1976,6 +2022,21 @@ describe(`StargateMongoose - ${testClientName} Connection - collections.collecti
       const docs = await collection.find({}, { sort: { username: 1 } }).toArray();
       assert.deepStrictEqual(docs.map(doc => doc.answer), [undefined, 42, undefined]);
     });
+    it.skip('findOneAndReplace should make _id an ObjectId when upserting with no _id', async () => {
```

Comment:
Should we implement support for `upsert` in `findOneAndReplace`? `upsert` doesn't seem to be in the options list for `findOneAndReplace` API as per doc here https://mongoosejs.com/docs/api/model.html#Model.findOneAndReplace()

Reply:
I need to add `upsert` to the Mongoose docs, but [here's the MongoDB types that show that `upsert` is a supported option for `findOneAndReplace()`](https://github.com/mongodb/node-mongodb-native/blob/ee56c8eebf22b2f90885513e699d209c63379846/src/operations/find_and_modify.ts#L40-L52).

`upsert` support for `findOneAndReplace()` is a nice to have. Not critical, but would be great to add if we have capacity.

Reply:
Got it. But, in the case of `findOneAndReplace`, the function takes only the replacement document and not the `update` operator (like how it is accepted in `findOneAndUpdate`, `updateOne` and `updateMany`), so there is no param to send "$setOnInsert" and it always be `ObjectId`.

Reply:
@vkarpov15 I just noticed that we take the `_id` from the `replacement` document in the case of `findOneAndReplace`. So I think we are good on this.

Reply:
Let me enable and run these skipped tests now and see if it looks good since there was a JSON API release last week.

Reply:
Tests look good.